Zig no longer compiles on FreeBSD

### Zig Version

0.11.0-dev.4248+48e0b89f2

### Steps to Reproduce and Observed Behavior

```zig
$ zig build -Dstatic-llvm --search-prefix "/usr/local/llvm16" --zig-lib-dir lib
lib/std/process.zig:1171:36: error: expected type 'error{UnknownTotalSystemMemory}', found 'error{SystemResources,Unexpected,PermissionDenied}'
                else => |e| return e,
                                   ^
lib/std/process.zig:1171:36: note: 'error.SystemResources' not a member of destination error set
lib/std/process.zig:1171:36: note: 'error.Unexpected' not a member of destination error set
lib/std/process.zig:1171:36: note: 'error.PermissionDenied' not a member of destination error set
referenced by:
    main: lib/build_runner.zig:316:30
    callMain: lib/std/start.zig:588:32
```

### Expected Behavior

It used to compile successfully in this way

I think this is probably related to the recent audit #16636

--- a/lib/std/process.zig
+++ b/lib/std/process.zig
@@ -1168,7 +1168,7 @@ pub fn totalSystemMemory() TotalSystemMemoryError!usize {
             var len: usize = @sizeOf(c_ulong);
             os.sysctlbynameZ("hw.physmem", &physmem, &len, null, 0) catch |err| switch (err) {
                 error.NameTooLong, error.UnknownName => unreachable,
-                else => |e| return e,
+                else => return error.UnknownTotalSystemMemory,
             };
             return @as(usize, @intCast(physmem));
         },

